src/components/UriViewer/LogViewer.ts

```diff
@@ -0,0 +1,231 @@
+import { LoadedState } from '@terra-ui-packages/core-utils';
+import _ from 'lodash/fp';
+import { useCallback, useEffect, useState } from 'react';
+import { dd, div, dl, dt, h, span } from 'react-hyperscript-helpers';
+import { ButtonOutline } from 'src/components/common';
+import { centeredSpinner, icon } from 'src/components/icons';
+import Modal from 'src/components/Modal';
+import { InfoBox } from 'src/components/PopupTrigger';
+import { SimpleTabBar } from 'src/components/tabBars';
+import { isAzureUri } from 'src/components/UriViewer/uri-viewer-utils';
+import { Ajax } from 'src/libs/ajax';
+import { useCancellation } from 'src/libs/react-utils';
+import { newTabLinkProps } from 'src/libs/utils';
+/**
+ * Information needed to preview a log file.
+ * @member logUri - The URI of the log file. Must be a valid Azure blob URI. No Sas token should be appended: a fresh one will be obtained.
+ * @member logTitle - The title of the log. Displayed to the user as a tab title.
+ * @member logKey - A unique key for this particular log.
+ * @member logFilename - The filename of this particular log. Does not need to be unique.
+ */
+export type LogInfo = {
+  logUri: string;
+  logTitle: string;
+  logKey: string;
+  logFilename: string;
+};
+
+/**
+ * Props for the LogViewer component.
+ * @member modalTitle - Should represent the group of logs being displayed. E.g. "Task Logs"
+ * @member logs - An array of LogInfo objects.
+ */
+export type LogViewerProps = {
+  modalTitle: string;
+  logs: LogInfo[];
+  onDismiss: () => void;
+};
+
+/**
+ * Props for the SimpleTab component. See src/components/tabBars.js
+ */
+type SimpleTabProps = {
+  key: string;
+  title: string;
+  width: number;
+};
+
+type FetchedLogData = {
+  textContent: string | undefined; // The literal characters of the log file.
```

Comment:
Nit: this file mixes two styles for documenting types.

IDK about IntelliJ, but from what I've found, the style that works best with VS Code and provides documentation in popups is:
```ts
/**
 * Docstring for type
 */
type MyType = {
  /** Docstring for field. */
  field: FieldType;

  ...
};